### ISSUE TITLE
can't find switch jump table in drivers/ata/sata_dwc_460ex.c

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#485]
```
$ ./scripts/config -e CONFIG_SATA_DWC
```

```
drivers/ata/sata_dwc_460ex.o: warning: objtool: sata_dwc_bmdma_start_by_tag()+0x3a2: can't find switch jump table
```

**Comment by nickdesaulniers** [comment]
hmm...seems like I only see it with allyesconfig...

**Comment by ihalip** [comment]
I'm seeing this with clang-8, but adding `-fno-jump-tables` to the compiler cmdline makes the warning goes away.

When building with GCC, -fno-jump-tables is added [here](https://github.com/ClangBuiltLinux/linux/blob/7c21383f3429dd70da39c0c7f1efa12377a47ab6/arch/x86/Makefile#L226).

**Comment by nickdesaulniers** [comment]
So @arndb and Jann may have a fix for this: https://lore.kernel.org/r/CAG48ez3ipuPHLxbqqc50=Kn4QuoNczkd7VqEoLPVd3WWLk2s+Q@mail.gmail.com/

**Comment by nathanchance** [comment]
Still relevant with tip of tree LLVM and Linux:

```
drivers/ata/sata_dwc_460ex.o: warning: objtool: sata_dwc_bmdma_start_by_tag()+0x3a2: can't find switch jump table
```

I will try to creduce something later.

**Comment by jpoimboe** [comment]
Potential fix: https://git.kernel.org/pub/scm/linux/kernel/git/jpoimboe/linux.git/commit/?h=objtool-clang-switch-fix&id=c796eeb0b88f95b2de9a4041a268f7f7d54735c6

**Comment by nickdesaulniers** [comment]
fixes: https://lore.kernel.org/r/cover.1581997059.git.jpoimboe@redhat.com/
